backend/danswer/background/celery/apps/app_base.py

```diff
@@ -157,34 +170,52 @@ def wait_for_redis(sender: Any, **kwargs: Any) -> None:
 
 
 def on_secondary_worker_init(sender: Any, **kwargs: Any) -> None:
-    r = get_redis_client()
-
     WAIT_INTERVAL = 5
     WAIT_LIMIT = 60
 
     logger.info("Running as a secondary celery worker.")
-    logger.info("Waiting for primary worker to be ready...")
+    logger.info("Waiting for all tenant primary workers to be ready...")
```

Comment:
All of the code should be a no-op for single tenant case, and have made a note to make this more robust / scale the primary worker locks across tenants